integration_tests/src/main/python/udf_test.py

```diff
@@ -214,6 +213,22 @@ def pandas_filter(iterator):
             conf=arrow_udf_conf)
 
 
+@pytest.mark.parametrize('data_gen', data_gens_nested_for_udf, ids=idfn)
+def test_pandas_map_udf_nested_type(data_gen):
+    # Spark supports limited types as the return type of pandas UDF.
+    # For details please go to
+    #  https://github.com/apache/spark/blob/master/python/pyspark/sql/pandas/types.py#L28
+    # So we return the data frames with one column of integral type for all types of the input.
```

Comment:
The return types don't look _that_ limited.  It supports strings, arrays (as long as they don't contain structs or timestamps), structs that don't contain other structs, NullType, DecimalType, etc.  Looks like we're missing a lot of test cases here.

Reply:
Yes, @revans2  Should we have tests to cover all the output types supported by PySpark ?

Reply:
I updated the test for the supported output types check.